### PR TITLE
42 pos analysis

### PR DESCRIPTION
**Pull request by djyldyz**
This PR is linked to issue #42. It includes code related to POS analysis and phrase matching in the Guardian articles.

To enable running the example notebook from scratch, I have uploaded the input files to this Google Drive [folder](https://drive.google.com/drive/folders/1cEo2J5pTa_mKTI2GX6qfk2UVFDcqdqTX?usp=sharing).

---

Checklist:

- [ ] I have refactored my code out from `notebooks/`
- [x] I have checked the code runs
- [x] I have tested the code
- [x] I have run `pre-commit` and addressed any issues not automatically fixed
- [x] I have merged any new changes from `dev`
- [x] I have documented the code
  - [x] Major functions have docstrings
  - [ ] Appropriate information has been added to `README`s
- [ ] I have explained the feature in this PR or (better) in `output/reports/`
- [x] I have requested a code review

### CODE REVIEW

**Comment by djyldyz** [comment]
@beingkk Hi Karlis, I have revisited this PR and made some small tweaks. It's now ready for you to review. Happy to chat about any questions.

**Comment by beingkk** [comment]
Hi @djyldyz, thanks so much for this. I'll go through the PR in the next few days. 

Right off the bat, I tried running tests, but got this error:
```
FileNotFoundError: [Errno 2] No such file or directory: '/Users/karliskanders/Documents/innovation_sweet_spots/outputs/data/discourse_analysis_outputs/sentence_sample.pkl'
```

I couldn't find the file in the google drive, but perhaps I'm missing something?

File: innovation_sweet_spots/analysis/prototyping/public_discourse_analysis/pd_data_processing_utils.py [hunk 1]
```diff
@@ -226,4 +226,27 @@ def view_collocations(grouped_sentences, metadata_dict, sentence_record_dict,
...
\ No newline at end of file
+            print('----------')
+
+
+def view_collocations_given_year(year_sentences, metadata_dict, sentence_record_dict, 
+                      url_field = 'webUrl', title_field = 'webTitle'):
+    """Prints sentences and corresponding article metadata for a given year.
+    
+    Args:
+        grouped_sentences: A pandas dataframe.
```
Comment (beingkk) [inline innovation_sweet_spots/analysis/prototyping/public_discourse_analysis/pd_data_processing_utils.py]:
```suggestion
        year_sentences: A pandas dataframe.
```

**Comment by beingkk** [comment]
Thank you very much for this, Jyl - I will go ahead and merge this now, to then slightly refactor further and merge into `dev`
NB: Don't delete the branch, just in case I've messed something up during the merging process.

**Pull request merged**